src/shared/updateArray.js

```diff
@@ -0,0 +1,11 @@
+export const updateMultiDimensionalArray = (array, key, id, elementToBeAdded) => {
+  let newArr = Object.assign({}, array);
+  newArr[key][id] = elementToBeAdded;
+  return newArr;
+}
+
+export const updateSingleDimensionalArray = (array, key, elementToBeAdded) => {
+  let newArr = Object.assign({}, array);
+  newArr[key] = elementToBeAdded;
+  return newArr;
+}
```

Comment:
You're a good person, Eliot.

Reply:
Haha so this is what you were commenting on. I try my best! 🤓